### PR TITLE
feat: adds SimpleForm as the default way to implement a form with Beagle

### PR DESCRIPTION
**Pull request by diogooliveira-zup**
## Description

- add a new component `SimpleForm` (`beagle:simpleform`). It should be rendered just like a container in mobile apps and as the html tag `<form>` in web applications. `SimpleForm` receives the same properties as `Container` and one more: `onSubmit`, which is an event (can be associated with an action or array of actions) that is triggered every time the form is submitted.

- Add a new default action named `SubmitForm` (`beagle:submitForm`). This action should not receive any additional parameter and its only function is to submit the parent form.

- the older `Form` component (`beagle:form`) continues to exist, but it will be **deprecated** in V1.1 and removed in the future.

## Related Issues

https://github.com/ZupIT/beagle/issues/54#issuecomment-634694003, #463

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [DCO].
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/main/java/br/com/zup/beagle/android/action/SubmitForm.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.android.action
+
+import br.com.zup.beagle.android.widget.RootView
+
+
+class SubmitForm() : Action{
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/action/SubmitForm.kt:22]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/9d0b3e83c09c2e52c93a75f96c55ac3d2780a456/android/beagle/src/main/java/br/com/zup/beagle/android/action/SubmitForm.kt#L22"></span>An empty default constructor can be removed.</td>
    </tr>
  </tbody>
</table>

File: backend/backend-widgets/src/main/kotlin/br/com/zup/beagle/widget/action/SubmitForm.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.widget.action
+
+class SubmitForm() : Action
```
Comment (botbeaglezup) [inline backend/backend-widgets/src/main/kotlin/br/com/zup/beagle/widget/action/SubmitForm.kt:19]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/9d0b3e83c09c2e52c93a75f96c55ac3d2780a456/backend/backend-widgets/src/main/kotlin/br/com/zup/beagle/widget/action/SubmitForm.kt#L19"></span>An empty default constructor can be removed.</td>
    </tr>
  </tbody>
</table>

File: android/beagle/src/main/java/br/com/zup/beagle/android/action/SubmitForm.kt (hunk 1 above)
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/action/SubmitForm.kt:22]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/9d0b3e83c09c2e52c93a75f96c55ac3d2780a456/android/beagle/src/main/java/br/com/zup/beagle/android/action/SubmitForm.kt#L22"></span>An empty default constructor can be removed.</td>
    </tr>
  </tbody>
</table>

File: backend/backend-widgets/src/main/kotlin/br/com/zup/beagle/widget/action/SubmitForm.kt (hunk 2 above)
Comment (botbeaglezup) [inline backend/backend-widgets/src/main/kotlin/br/com/zup/beagle/widget/action/SubmitForm.kt:19]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/9d0b3e83c09c2e52c93a75f96c55ac3d2780a456/backend/backend-widgets/src/main/kotlin/br/com/zup/beagle/widget/action/SubmitForm.kt#L19"></span>An empty default constructor can be removed.</td>
    </tr>
  </tbody>
</table>

File: android/beagle/src/main/java/br/com/zup/beagle/android/action/SimpleFormSubmitAction.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.android.action
+
+import br.com.zup.beagle.android.components.utils.submitSimpleForm
+import br.com.zup.beagle.android.widget.RootView
+
+class SimpleFormSubmitAction : Action {
```
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/action/SimpleFormSubmitAction.kt:22]: It's this name the same in Android, iOS and Web?

File: android/beagle/src/main/java/br/com/zup/beagle/android/action/SimpleFormSubmitAction.kt [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    override fun execute(rootView: RootView) {
+        rootView.submitSimpleForm(rootView)
```
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/action/SimpleFormSubmitAction.kt:25]: I think this should not be a extension or a method inside rootView, this should be a logic inside this action

File: android/beagle/src/main/java/br/com/zup/beagle/android/action/SubmitForm.kt [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.android.action
+
+import br.com.zup.beagle.android.widget.RootView
+
+class SubmitForm : Action{
```
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/action/SubmitForm.kt:21]: Duplicated?

File: android/beagle/src/main/java/br/com/zup/beagle/android/components/Button.kt [hunk 6]
```diff
@@ -69,6 +71,8 @@ data class Button(
 
         button.setOnClickListener {
             onPress?.let {
+                (rootView as BaseRootView).currentViewAction = button
```
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/components/Button.kt:74]: Did you miss to remove this code?

File: android/beagle/src/main/java/br/com/zup/beagle/android/components/form/SimpleForm.kt [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+    private fun addChildrenForm(beagleFlexView: BeagleFlexView, rootView: RootView){
+        children.forEach{child ->
+            beagleFlexView.addServerDrivenComponent(child, rootView)
+        }
+    }
+
+    fun submit(rootView: RootView) {
+        onSubmit.forEach { action ->
+            action.execute(rootView)
```
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/components/form/SimpleForm.kt:75]: The correct way is not call `action.execute` it's `handleEvent`

File: android/beagle/src/main/java/br/com/zup/beagle/android/components/utils/FormExtensions.kt [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+package br.com.zup.beagle.android.components.utils
+
+import android.view.ViewGroup
+import br.com.zup.beagle.android.components.form.SimpleForm
+import br.com.zup.beagle.android.engine.renderer.BaseRootView
+import br.com.zup.beagle.android.widget.RootView
+
+fun RootView.submitSimpleForm(rootView: RootView) = this.apply {
```
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/components/utils/FormExtensions.kt:24]: Move this logic to Action instead.

File: android/beagle/src/main/java/br/com/zup/beagle/android/engine/renderer/BaseRootView.kt [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.android.engine.renderer
+
+import android.view.View
+import br.com.zup.beagle.android.widget.RootView
+
+abstract class BaseRootView : RootView {
```
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/engine/renderer/BaseRootView.kt:22]: Does this class is necessary, this could not be a simple attribute inside RootView?

File: android/beagle/src/main/java/br/com/zup/beagle/android/utils/WidgetExtensions.kt [hunk 10]
```diff
@@ -93,6 +103,7 @@ fun ServerDrivenComponent.toView(activity: AppCompatActivity) = this.toView(Acti
  */
 fun ServerDrivenComponent.toView(fragment: Fragment) = this.toView(FragmentRootView(fragment))
 
+
```
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/utils/WidgetExtensions.kt:106]: Remove this extra space.

File: android/beagle/src/main/java/br/com/zup/beagle/android/utils/WidgetExtensions.kt [hunk 11]
```diff
@@ -79,6 +80,15 @@ fun <T> ServerDrivenComponent.observeBindChanges(
     bind.observe(rootView, observes)
 }
 
+/**
+ * Transform your Component to a view.
+ * @property rootView the contract that is propagated from create view tree algorithm.
+ * @property view the view that triggered the action.
+ */
+fun ServerDrivenComponent.currentViewTriggerAction(rootView: RootView, view: View){
```
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/utils/WidgetExtensions.kt:88]: As we talked, this should not be exposed, all the logic to save the RootView should be handled inside the `BeagleFlexView`.

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/custom/BeagleFlexView.kt [hunk 12]
```diff
@@ -53,7 +56,13 @@ internal open class BeagleFlexView(
             serverDrivenComponent
         }
         val style = (component as? StyleComponent)?.style ?: Style()
-        val view = viewRendererFactory.make(serverDrivenComponent).build(rootView)
+        val rootViewCopy: BaseRootView?
```
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/view/custom/BeagleFlexView.kt:59]: I think you miss the code to save the current view before to make a copy.

File: android/beagle/src/test/java/br/com/zup/beagle/android/components/form/SimpleFormTest.kt [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+
+    override fun setUp() {
+        super.setUp()
+
+        simpleForm = SimpleForm(context, onSubmit, children)
+        every { simpleFormAction.execute(rootView) } just Runs
+    }
+
+    @Test
+    fun build_should_return_a_beagle_flex_view_instance() {
```
Comment (tuliopereirazup) [inline android/beagle/src/test/java/br/com/zup/beagle/android/components/form/SimpleFormTest.kt:46]: What do you think to create a test to check if every child is created and added to the view group?

File: backend/backend-widgets/src/main/kotlin/br/com/zup/beagle/widget/form/SimpleForm.kt [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+package br.com.zup.beagle.widget.form
+
+import br.com.zup.beagle.core.ServerDrivenComponent
+import br.com.zup.beagle.widget.action.Action
+import br.com.zup.beagle.widget.context.ContextData
+
+class SimpleForm (
+    val context: ContextData,
```
Comment (tuliopereirazup) [inline backend/backend-widgets/src/main/kotlin/br/com/zup/beagle/widget/form/SimpleForm.kt:24]: ContextData should not be used as normal parameter, it should inherit from `ContextDataComponent`

File: android/beagle/src/main/java/br/com/zup/beagle/android/action/SimpleFormSubmitAction.kt (hunk 3 above)
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/action/SimpleFormSubmitAction.kt:22]: Create a test case for this class as well.

File: android/beagle/src/main/java/br/com/zup/beagle/android/action/SubmitForm.kt [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.android.action
+
+import br.com.zup.beagle.android.components.utils.submitSimpleForm
+import br.com.zup.beagle.android.widget.RootView
+
+class SubmitForm : Action{
```
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/action/SubmitForm.kt:22]: Create a test case for this class.

**Comment by theffc** [comment]
Shouldn't this PR be named something like`feat: adds SimpleForm as the default way to implement a form with Beagle`? Because it seems that we have a complete implementation of SimpleForm here (on both platforms), instead of just a fix on android to make it work

**Comment by uziasferreirazup** [comment]
@theffc I will get this PR and add deprecated in these components and change the title of PR

File: backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/SimpleFormScreenBuilder.kt [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+
+package br.com.zup.beagle.sample.builder
+
+import br.com.zup.beagle.widget.action.Alert
+import br.com.zup.beagle.widget.action.SubmitForm
+import br.com.zup.beagle.widget.context.ContextData
+import br.com.zup.beagle.widget.core.ScrollAxis
+import br.com.zup.beagle.widget.core.TextInputType
+import br.com.zup.beagle.widget.form.SimpleForm
+import br.com.zup.beagle.widget.layout.*
```
Comment (botbeaglezup) [inline backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/SimpleFormScreenBuilder.kt:25]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/52a729fa4aecabc55aae44b67c5d31e5e215e2dd/backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/SimpleFormScreenBuilder.kt#L25"></span>br.com.zup.beagle.widget.layout.* is a wildcard import. Replace it with fully qualified imports.</td>
    </tr>
  </tbody>
</table>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/52a729fa4aecabc55aae44b67c5d31e5e215e2dd/android/beagle/src/main/java/br/com/zup/beagle/android/context/ContextActionExecutor.kt#L27">android/beagle/src/main/java/br/com/zup/beagle/android/context/ContextActionExecutor.kt#L27</a> - The function executeActions(rootView: RootView, origin: View, sender: Any, actions: List<Action>, eventName: String, eventValue: Any?) has too many parameters. The current threshold is set to 6.</Action></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_backend">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/SimpleFormScreenBuilder.kt (hunk 16 above)
Comment (botbeaglezup) [inline backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/SimpleFormScreenBuilder.kt:25]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/52a729fa4aecabc55aae44b67c5d31e5e215e2dd/backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/SimpleFormScreenBuilder.kt#L25"></span>br.com.zup.beagle.widget.layout.* is a wildcard import. Replace it with fully qualified imports.</td>
    </tr>
  </tbody>
</table>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/52a729fa4aecabc55aae44b67c5d31e5e215e2dd/android/beagle/src/main/java/br/com/zup/beagle/android/context/ContextActionExecutor.kt#L27">android/beagle/src/main/java/br/com/zup/beagle/android/context/ContextActionExecutor.kt#L27</a> - The function executeActions(rootView: RootView, origin: View, sender: Any, actions: List<Action>, eventName: String, eventValue: Any?) has too many parameters. The current threshold is set to 6.</Action></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/c08709ab8271c1806134d8f80a5db71fd5c61543/android/beagle/src/main/java/br/com/zup/beagle/android/utils/ToolbarManager.kt#L178">android/beagle/src/main/java/br/com/zup/beagle/android/utils/ToolbarManager.kt#L178</a> - Prefer “<code>SHOW_AS_ACTION_IF_ROOM</code>” instead of “<code>SHOW_AS_ACTION_ALWAYS</code>”</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/main/java/br/com/zup/beagle/android/action/SubmitForm.kt [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+            if (currentView is ViewGroup && currentView.tag is SimpleForm) {
+                (currentView.tag as SimpleForm).submit(rootView, origin)
+                foundSimpleForm = true
+                break
+            }
+            currentView = currentView.parent
+        }
+
+        if (!foundSimpleForm) {
+            BeagleLoggerProxy.error("not found simple form in the parents")
```
Comment (matheusribeirozup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/action/SubmitForm.kt:41]: Instead of using the `BeagleLoggerProxy` directly, create the error message in the `BeagleMessageLogs` object and log it there

File: android/beagle/src/main/java/br/com/zup/beagle/android/components/form/core/Constants.kt [hunk 18]
```diff
@@ -16,6 +16,7 @@
 
 package br.com.zup.beagle.android.components.form.core
 
+@Deprecated("use SimpleForm and SubmitForm instead")
```
Comment (matheusribeirozup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/components/form/core/Constants.kt:19]: What do you think about extracting this message into a variable?

File: android/beagle/src/main/java/br/com/zup/beagle/android/action/FormLocalAction.kt [hunk 19]
```diff
@@ -29,12 +33,12 @@ data class FormLocalAction(
...
     var formLocalActionHandler: FormLocalActionHandler? = BeagleEnvironment.beagleSdk.formLocalActionHandler
 
-    override fun execute(rootView: RootView) {
+    override fun execute(rootView: RootView, origin: View) {
         formLocalActionHandler?.handle(rootView.getContext(), this, object : ActionListener {
 
             override fun onSuccess(action: Action) {
                 changeActivityState(rootView, ServerDrivenState.Loading(false))
-                action.execute(rootView)
+                action.execute(rootView, origin)
```
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/action/FormLocalAction.kt:41]: Use `handleEvent` instead

File: android/beagle/src/main/java/br/com/zup/beagle/android/components/Button.kt [hunk 20]
```diff
@@ -67,9 +69,10 @@ data class Button(
             viewFactory.makeButton(rootView.getContext(), getStyleId(this.styleId))
         else viewFactory.makeButton(rootView.getContext())
 
-        button.setOnClickListener {
+        button.setOnClickListener { view ->
             onPress?.let {
-                this@Button.handleEvent(rootView, it, "onPress")
+                this@Button.currentViewTriggerAction(rootView, button)
```
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/components/Button.kt:74]: Remove this line, as this is part of old solution.

File: android/beagle/src/main/java/br/com/zup/beagle/android/engine/renderer/BaseRootView.kt (hunk 9 above)
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/engine/renderer/BaseRootView.kt:22]: Delete this class as this is part of old solution.

File: android/beagle/src/main/java/br/com/zup/beagle/android/engine/renderer/RootView.kt [hunk 21]
```diff
@@ -20,19 +20,20 @@ import android.content.Context
 import androidx.appcompat.app.AppCompatActivity
 import androidx.fragment.app.Fragment
 import androidx.lifecycle.LifecycleOwner
-import br.com.zup.beagle.android.widget.RootView
 
-class FragmentRootView(
+data class FragmentRootView(
```
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/engine/renderer/RootView.kt:24]: Revert this changes on this class, as this is part of old solution

File: android/beagle/src/main/java/br/com/zup/beagle/android/utils/WidgetExtensions.kt [hunk 22]
```diff
@@ -79,6 +84,15 @@ fun <T> ServerDrivenComponent.observeBindChanges(
     bind.observe(rootView, observes)
 }
 
+/**
+ * Transform your Component to a view.
+ * @property rootView the contract that is propagated from create view tree algorithm.
+ * @property view the view that triggered the action.
+ */
+fun ServerDrivenComponent.currentViewTriggerAction(rootView: RootView, view: View) {
```
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/utils/WidgetExtensions.kt:92]: Remove this code, as this is part of old solution.

File: android/beagle/src/main/java/br/com/zup/beagle/android/action/SubmitForm.kt [hunk 23]
```diff
@@ -0,0 +1,45 @@
...
+import br.com.zup.beagle.android.logger.BeagleLoggerProxy
+import br.com.zup.beagle.android.logger.BeagleMessageLogs
+import br.com.zup.beagle.android.widget.RootView
+
+class SubmitForm : Action {
+    override fun execute(rootView: RootView, origin: View) {
+        var currentView: ViewParent? = origin.parent
+
+        var foundSimpleForm = false
+        while (currentView != null) {
```
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/action/SubmitForm.kt:32]: What do you think to change the algorithm to this?

```suggestion
        while (foundSimpleForm || currentView != null) {
             if (currentView is ViewGroup && currentView.tag is SimpleForm) {
                 foundSimpleForm = true
             } else {
                 currentView = currentView.parent
             }
         }
         
         if (foundSimpleForm) {
            (currentView.tag as SimpleForm).submit(rootView, origin)
         } else {
            BeagleMessageLogs.logNotFoundSimpleForm()
         }
```
